sql-plugin/src/main/scala/com/nvidia/spark/rapids/RmmRapidsRetryIterator.scala

```diff
@@ -474,10 +478,12 @@ object RmmRapidsRetryIterator extends Logging {
           s"Current threadCountBlockedUntilReady: ${threadCountBlockedUntilReady}. " +
           s"The current attempt: " +
           s"{${attemptStack.head}}"
-        if (isFromGpuOom) {
-          throw new GpuSplitAndRetryOOM(s"GPU OutOfMemory: $message")
-        } else {
-          throw new CpuSplitAndRetryOOM(s"CPU OutOfMemory: $message")
+        splitReason match {
+          case SplitReason.GPU_OOM => throw new GpuSplitAndRetryOOM(s"GPU OutOfMemory: $message")
+          case SplitReason.CPU_OOM => throw new CpuSplitAndRetryOOM(s"CPU OutOfMemory: $message")
+          case SplitReason.OTHER =>
+            throw new IllegalStateException("Non OOM State Machine reason caused split," +
```

Comment:
I think the specific reason ("Column overflow") would be better than just "Non OOM State Machine reason". And many users are likely to not know "State Machine". Better not mention it.
  @GaryShen2008 Any input for this message.

Reply:
neither will they understand "CUDF overflow" or even "split and retry", it's just some useful information to the troubleshooter with domain knowledge.

Reply:
Column overflow seems just one of the cases, not the only one case. Am I right?
If so, we can't just say the column overflow.

Here I think it's ok if we can't know the exact reason caused split. If we have multiple cases caused the exception here, we'd better to log it more clear.

Reply:
yes, the detail of the exception is logged a bit earlier. @GaryShen2008

Reply:
I am OK with OTHER if other guys agree with it.